rust-runtime/aws-smithy-http-server/src/protocols.rs

```diff
@@ -35,6 +30,16 @@ pub fn content_type_header_empty_body_no_modeled_input<B>(
     }
 }
 
+fn parse_content_type(headers: &HeaderMap) -> Result<mime::Mime, MissingContentTypeReason> {
+    headers
+        .get(http::header::CONTENT_TYPE)
+        .unwrap() // The header is present, `unwrap` will not panic.
+        .to_str()
+        .map_err(MissingContentTypeReason::ToStrError)?
+        .parse::<mime::Mime>()
+        .map_err(MissingContentTypeReason::MimeParseError)
+}
+
```

Comment:
IntelliJ was complaining about code duplication here so I extracted the common code